src/lib/draw/route/SplitRouteMode.svelte

```diff
@@ -86,56 +84,41 @@
     if (snappedIndex == null) {
       // We clicked the map, stop the tool
       mode.set({ mode: "list" });
-    } else {
-      // TODO Can we avoid using ! everywhere here?
-      let result = lineSplit(
-        $gjSchemeCollection.features[snappedIndex!] as Feature<LineString>,
-        snappedCursor!,
-      );
-      if (result.features.length == 2) {
-        let piece1 = result.features[0];
-        let piece2 = result.features[1];
-        // lineSplit may introduce unnecessary coordinate precision
-        piece1.geometry.coordinates =
-          piece1.geometry.coordinates.map(setPrecision);
-        piece2.geometry.coordinates =
-          piece2.geometry.coordinates.map(setPrecision);
-
-        gjSchemeCollection.update((gj) => {
-          // Keep the old ID for one, assign a new ID to the other
-          piece1.id = gj.features[snappedIndex!].id;
-          piece2.id = newFeatureId(gj);
-
-          // The properties get lost. Copy everything to both
-          piece1.properties = JSON.parse(
-            JSON.stringify(gj.features[snappedIndex!].properties),
-          );
-          // "Deep clone"
-          piece2.properties = JSON.parse(JSON.stringify(piece1.properties));
-
-          fixRouteProperties(
-            gj.features[snappedIndex!] as OurFeature<LineString>,
-            piece1 as OurFeature<LineString>,
-            piece2 as OurFeature<LineString>,
-            snappedCursor!,
-          );
-
-          // Replace the one LineString we snapped to with the two new pieces
-          gj.features.splice(
-            snappedIndex!,
-            1,
-            piece1 as OurFeature<LineString>,
-            piece2 as OurFeature<LineString>,
-          );
-
-          return gj;
-        });
-      }
+      return;
+    }
 
-      // Stay in this mode, but reset state
-      snappedCursor = null;
-      snappedIndex = null;
+    let result = splitRoute(
+      $gjSchemeCollection.features[snappedIndex] as unknown as Feature<
+        LineString,
+        RouteProps
+      >,
+      snappedCursor!,
+    );
+
+    if (result != null) {
+      let [piece1, piece2] = result;
+      gjSchemeCollection.update((gj) => {
+        // Keep the old ID for one, assign a new ID to the other
+        piece1.id = gj.features[snappedIndex!].id;
+        piece2.id = newFeatureId(gj);
+
+        // Replace the one LineString we snapped to with the two new pieces
+        gj.features.splice(
+          snappedIndex!,
+          1,
+          // TODO Maybe splitRoute's types can be more precise. Existing
+          // properties get copied, so this is valid.
+          piece1 as unknown as OurFeature<LineString>,
+          piece2 as unknown as OurFeature<LineString>,
+        );
+
+        return gj;
```

Comment:
does returning the `gj` do anything? Looks to me like it may as well be `return;` since we're editing in place.

Reply:
https://svelte.dev/docs/svelte-store#writable
It's necessary. `update is a method that takes one argument which is a callback. The callback takes the existing store value as its argument and returns the new value to be set to the store.`

Reply:
ahhh yes didn't spot we were in a callback cheers!
LGTM